src/hotspot/share/gc/g1/g1BiasedArray.cpp

```diff
@@ -26,11 +26,31 @@
 #include "gc/g1/g1BiasedArray.hpp"
 #include "memory/padded.inline.hpp"
 
+G1BiasedMappedArrayBase::G1BiasedMappedArrayBase() :
+  _alloc_base(NULL),
+  _base(NULL),
+  _length(0),
+  _biased_base(NULL),
+  _bias(0),
+  _shift_by(0) { }
+
+G1BiasedMappedArrayBase::~G1BiasedMappedArrayBase() {
+  if (_alloc_base != NULL) {
+    FREE_C_HEAP_ARRAY(u_char, _alloc_base);
+    _alloc_base = NULL;
+    _base = NULL;
+    _length = 0;
+    _biased_base = NULL;
+    _bias = 0;
+    _shift_by = 0;
+  }
```

Comment:
1. `FreeHeap` makes more sense to me, since the allocation sites uses `AllocateHeap`.
2. Why resetting those fields to zero? The only resource we need to release in this destructor is memory, right?

Reply:
1. Will fix.
2. Some debugging code to see if anyone else is using it after the freeing as previously these tables were never freed - I was not sure if I should keep them or not so I want to see if anyone would complain :) I'll remove these then.